RELEASE-NOTES.md

```diff
@@ -1,5 +1,63 @@
 # Prefect Release Notes
 
+## Release 2.16.4
+
+### Flow Run Graph updates
+
+The Flow Run Graph has been updated to display additional layers of information! Interactive and real-time state changes and artifacts are now visible in context on the graph.
+
+<img width="892" alt="The Prefect flow run graph" src="https://github.com/PrefectHQ/prefect/assets/6776415/422a878e-f8bc-46b5-ae76-830f24861aaf">
+
+### Agents
+
+A year ago, we released workers as a replacement for agents. Workers significantly enhance the experience of deploying flows and simplify the specification of each flow's infrastructure and runtime environment.
+
+With this release we are adding a six month (September 14) deprecation warning to agents and related concepts. Please note that:
+
+- Deprecation will not impact or break any work running with agents and agent-related concepts - although we encourage users to upgrade to workers because they provide a better deployment experience, you can continue to use existing agents and related concepts after deprecation
+- After September 14, Prefect Cloud users will not be able to create new agent work pools or infrastructure blocks
+- After September 14, new minor versions of the Prefect Python package will not include agents
+
+Like agents, workers support creating deployments through the Prefect CLI and through Python, but require different syntax.
```

Comment:
Including a link to the upgrade guide will probably help some users.

```suggestion
Like agents, workers support creating deployments through the Prefect CLI and through Python, but require different syntax.

For more information, please refer to the [Upgrade from Agents to Workers](https://docs.prefect.io/latest/guides/upgrade-guide-agents-to-workers/) guide
```

Reply:
Nice!  Thank you